hypertrace-gradle-java-convention-plugin/src/main/java/org/hypertrace/gradle/java/convention/JavaConventionPlugin.java

```diff
@@ -0,0 +1,66 @@
+package org.hypertrace.gradle.java.convention;
+
+import java.util.List;
+import java.util.Objects;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.Task;
+import org.gradle.api.plugins.JavaPluginExtension;
+import org.gradle.jvm.toolchain.JavaToolchainService;
+import org.gradle.jvm.toolchain.JavaToolchainSpec;
+
+public class JavaConventionPlugin implements Plugin<Project> {
+
+  public void apply(Project target) {
+    target
+        .getPluginManager()
+        .withPlugin(
+            "java",
+            unused -> {
+              createConvention(target);
+              configureToolchain(target);
+              modifyTestJvmArgs(target);
+            });
+  }
+
+  private void createConvention(Project target) {
+    target
+        .getExtensions()
+        .create(JavaConventionExtension.EXTENSION_NAME, JavaConventionExtension.class);
+  }
+
+  private void configureToolchain(Project target) {
+    JavaToolchainSpec spec = javaPluginExtension(target).getToolchain();
+    spec.getLanguageVersion().set(javaConventionExtension(target).languageVersion);
+    javaToolchainService(target).compilerFor(spec);
+    javaToolchainService(target).launcherFor(spec);
+    javaToolchainService(target).javadocToolFor(spec);
+  }
+
+  private void modifyTestJvmArgs(Project target) {
+    Task testTask = target.getTasks().findByName("test");
+    if (Objects.nonNull(testTask)) {
+      // required for junit environment variables for jdk 17+
+      // https://junit-pioneer.org/docs/environment-variables/#warnings-for-reflective-access
+      testTask.setProperty(
```

Comment:
prefer using the type safe accessors - instead of using find by name you can use (assuming we want this for any test task like integration tests too)

```java
    target.getTasks().withType(Test.class).configureEach(task -> {
      task.jvmArgs(...)
    });
```

If you specifically want that task:
```java
    target.getTasks().named(JavaPlugin.TEST_TASK_NAME, Test.class).configure(task -> {
      task.jvmArgs(...)
    });
```

Reply:
Yes, we would want them for integration too. Thanks, will use type safe. Is there any way to figure out the task class hierarchy from task name?

Reply:
Do you mean like you know a task name and want to figure out what type to use while developing? You can do what you did here and then inspect the result of findByName to see what the impl is. Of course, you can also go look at the code if you know what plugin it originates from.

Reply:
Yes I meant that. Is there any way to inspect / attach debugger for plugins? Currently I am using lifecycle loggers to debug everything

Reply:
> Is there any way to inspect / attach debugger for plugins?

Yes - you can use the `-Dorg.gradle.debug=true` flag to start up gradle in debug mode and then attach to it (also suggest using `--no-daemon` when debugging).